src/helpers.ts

```diff
@@ -198,3 +198,12 @@ interface SurvivalGameInfo {
      */
     wave: number;
 }
+
+interface _Constructor<T> {
+    readonly prototype: T;
```

Comment:
As it turns out, actually -- many of the things I thought were instance-fields are actually self-memoizing properties, see:

```javascript
    utils.defineGameObjectProperties(Creep.prototype, data, {
        name: (o) => o.user == runtimeData.user._id ? o.name : undefined,
        body: (o) => o.body,
        ...
    });
```

Where `defineGameObjectProperties` is defined as follows:
```javascript
exports.defineGameObjectProperties = function(obj, dataFn, properties) {
    var propertiesInfo = {};

    for(var name in properties) {
        eval(`
            propertiesInfo['${name}'] = {
                enumerable: true,
                get() {
                    if(!this['_${name}']) {
                        this['_${name}'] = properties['${name}'](dataFn(this.id));
                    }
                    return this['_${name}'];
                }
            }`);
    }
    Object.defineProperties(obj, propertiesInfo);

    obj.toJSON = function() {
        var result = {};
        for(var i in this) {
            if(i[0] == '_' || _.contains(['toJSON','toString'],i)) {
                continue;
            }
            result[i] = this[i];
        }
        return result;
    }
};
```

I deem life too short to accurately represent the types of prototype objects.